src/passkey_child/passkey_child_credentials.c

```diff
@@ -137,6 +139,49 @@ prepare_credentials(struct passkey_data *data, fido_dev_t *dev,
     return ret;
 }
 
+errno_t
+passkey_recv_pin(TALLOC_CTX *mem_ctx, int fd, char **_pin)
+{
+    uint8_t buf[IN_BUF_SIZE];
+    ssize_t len;
+    errno_t ret;
+    char *str;
+
+    errno = 0;
+    len = sss_atomic_read_s(fd, buf, IN_BUF_SIZE);
+    if (len == -1) {
+        ret = errno;
+        ret = (ret == 0) ? EINVAL: ret;
+        DEBUG(SSSDBG_CRIT_FAILURE,
+              "read failed [%d][%s].\n", ret, strerror(ret));
+        return ret;
+    }
+
+    if (len == 0 || *buf == '\0') {
+        DEBUG(SSSDBG_CRIT_FAILURE, "Missing PIN.\n");
+        return EINVAL;
+    }
+
+    /* Remove the end of line '\n' character */
+    len--;
```

Comment:
Is this needed?  When calling `passkey_child --authenticate...` from SSSD I had to remove this line, or the trailing digit of the PIN was chopped off. I used `echo -n $PIN` when testing on the command line.

Reply:
Removed. Thanks for reporting the issue.